src/types/redis_stream_base.cc

```diff
@@ -85,39 +79,48 @@ Status ParseStreamEntryID(const std::string &input, StreamEntryID *id) {
   return Status::OK();
 }
 
-Status ParseNewStreamEntryID(const std::string &input, NewStreamEntryID *id) {
+std::unique_ptr<NextStreamEntryIDGenerationStrategy> ParseNextStreamEntryIDStrategy(const std::string &input) {
```

Comment:
From my side, I recommend you to use something like `StatusOr<std::unique_ptr<NextStreamEntryIDGenerationStrategy>>`, e.g.
```
StatusOr<std::unique_ptr<NextStreamEntryIDGenerationStrategy>> ParseNextStreamEntryIDStrategy(const std::string &input) {
  ...
  if (...) {
    return {Status::NotOk, "some reason..."};
  } else {
    return std::make_unique<SomeDerivedClass>(...);
  }
  ...
}
```